Tests/Library/Validator/ResolveValidatorTest.php

```diff
@@ -80,6 +81,24 @@ public function testValidFragmentTypeWithComposite()
       $validator->assertValidFragmentForField($fragment, $fragmentReference, new NonNullType($userType));
     }
 
+    public function testValidFragmentTypeWithUnion()
+    {
+      $validator = new ResolveValidator(new ExecutionContext(new TestSchema()));
+
+      $unionType = new ObjectType([
```

Comment:
You named the variable $unionType but put an ObjectType in there. You should not be creating an Object Type at all. Instead, all we need is a union type by itself which we can then pass to the assertValidFragmentForField method.